***

## Logistic Regression Classification

***

### 👉🏾Data Preprocessing

#### ☑️ Importing Dataset and Libraries

In [1]:
import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt

In [2]:
data = pd.read_csv('C:/Users/HP Elitebook/OneDrive/Bureau/HOMEWORKS/PYTHON/ML_BOOTCAMP/Machine Learning A-Z (Codes and Datasets)/Part 3 - Classification/Section 14 - Logistic Regression/Python/Social_Network_Ads.csv')
data

Age  EstimatedSalary  Purchased
0     19            19000          0
1     35            20000          0
2     26            43000          0
3     27            57000          0
4     19            76000          0
..   ...              ...        ...
395   46            41000          1
396   51            23000          1
397   50            20000          1
398   36            33000          0
399   49            36000          1

[400 rows x 3 columns]

#### ☑️ Setting Features

In [5]:
x = data.iloc[:, 0:-1].values
y = data.iloc[:, -1].values

#### ☑️ Splitting Dataset 

In [10]:
from sklearn.model_selection import train_test_split as tts
x_train, x_test, y_train, y_test = tts(x,y, test_size = 0.2, random_state = 0)

#### ☑️ Feature Scaling

In [49]:
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
x_train = sc.fit_transform(x_train)
x_test = sc.transform(x_test)

### 👉🏾 Training the LR on the training set 

In [16]:
from sklearn.linear_model import LogisticRegression
clf = LogisticRegression(random_state = 0)
clf.fit(x_train, y_train)

LogisticRegression(random_state=0)

### 👉🏾 Predictions


#### ☑️ Predicting for the first customer

In [52]:
clf.predict(sc.transform([[30, 87000]]))

array([1], dtype=int64)

#### ☑️ Predicting The test set

In [56]:
y_pred = clf.predict(x_test)

In [73]:
a = np.concatenate((y_test.reshape(len(y_test),1),y_pred.reshape(len(y_pred),1)),1)
a

array([[0, 0],
       [0, 0],
       [0, 0],
       [0, 0],
       [0, 0],
       [0, 0],
       [0, 0],
       [1, 1],
       [0, 0],
       [0, 1],
       [0, 0],
       [0, 0],
       [0, 0],
       [0, 1],
       [0, 0],
       [0, 1],
       [0, 0],
       [0, 0],
       [1, 1],
       [0, 0],
       [0, 0],
       [1, 1],
       [0, 0],
       [1, 1],
       [0, 0],
       [1, 1],
       [0, 0],
       [0, 0],
       [0, 0],
       [0, 0],
       [0, 0],
       [1, 0],
       [1, 1],
       [0, 0],
       [0, 0],
       [0, 0],
       [0, 0],
       [0, 0],
       [0, 0],
       [1, 1],
       [0, 0],
       [0, 0],
       [0, 0],
       [0, 0],
       [1, 1],
       [0, 0],
       [0, 0],
       [1, 1],
       [0, 0],
       [1, 1],
       [1, 1],
       [0, 0],
       [0, 0],
       [0, 1],
       [1, 1],
       [1, 0],
       [0, 0],
       [0, 0],
       [1, 1],
       [0, 0],
       [0, 0],
       [1, 1],
       [0, 0],
       [1, 1],
       [0, 0],
       [1, 1],
       [0,

### 👉🏾 Confusion Matrix and Accuracy Score

#### ☑️ Confusion Matrix

In [74]:
from sklearn.metrics import confusion_matrix as cm
conf = cm(y_test, y_pred)
conf

array([[52,  6],
       [ 3, 19]], dtype=int64)

So there's 52 correct prediction in the class of 0 ( customers that did not buy the SUV ).<br>
And 19 in the class of 1 ( customers that did buy the SUV ).

#### ☑️ Accuracy Score

In [76]:
from sklearn.metrics import accuracy_score as ac_
score = ac_(y_test, y_pred)
score

0.8875

This result means that 89% of our test set are correctly predicted